src/otx/core/data/entity/keypoint_detection.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTX detection data entities."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from torchvision import tv_tensors
+
+from otx.core.data.entity.base import (
+    OTXBatchDataEntity,
+    OTXBatchPredEntity,
+    OTXDataEntity,
+    OTXPredEntity,
+)
+from otx.core.data.entity.utils import register_pytree_node
+from otx.core.types.task import OTXTaskType
+
+if TYPE_CHECKING:
+    from torch import BoolTensor, LongTensor
+
+
+@register_pytree_node
+@dataclass
+class KeypointDetDataEntity(OTXDataEntity):
+    """Data entity for detection task.
+
+    :param bboxes: Bbox annotations as top-left-bottom-right
+        (x1, y1, x2, y2) format with absolute coordinate values
+    :param labels: Bbox labels as integer indices
+    """
+
+    @property
+    def task(self) -> OTXTaskType:
+        """OTX Task type definition."""
+        return OTXTaskType.KEYPOINT_DETECTION
+
+    bboxes: tv_tensors.BoundingBoxes
+    labels: LongTensor
+    keypoints: tv_tensors.TVTensor
+    keypoints_visible: BoolTensor
+    keypoint_x_labels: tv_tensors.TVTensor
+    keypoint_y_labels: tv_tensors.TVTensor
+    keypoint_weights: tv_tensors.TVTensor
+
+
+@dataclass
+class KeypointDetPredEntity(OTXPredEntity, KeypointDetDataEntity):
+    """Data entity to represent the detection model output prediction."""
+
+
+@dataclass
+class KeypointDetBatchDataEntity(OTXBatchDataEntity[KeypointDetDataEntity]):
+    """Data entity for detection task.
+
+    :param bboxes: A list of bbox annotations as top-left-bottom-right
+        (x1, y1, x2, y2) format with absolute coordinate values
+    :param labels: A list of bbox labels as integer indices
+    """
+
+    bboxes: list[tv_tensors.BoundingBoxes]
+    labels: list[LongTensor]
+    keypoints: list[tv_tensors.TVTensor]
+    keypoints_visible: list[BoolTensor]
+    keypoint_x_labels: list[tv_tensors.TVTensor]
+    keypoint_y_labels: list[tv_tensors.TVTensor]
+    keypoint_weights: list[tv_tensors.TVTensor]
+
+    @property
+    def task(self) -> OTXTaskType:
+        """OTX Task type definition."""
+        return OTXTaskType.DETECTION
+
+    @classmethod
+    def collate_fn(
+        cls,
+        entities: list[KeypointDetDataEntity],
+        stack_images: bool = True,
+    ) -> KeypointDetBatchDataEntity:
+        """Collection function to collect `DetDataEntity` into `DetBatchDataEntity` in data loader.
+
+        Args:
+            entities: List of `DetDataEntity`.
```

Comment:
Could you fix docstrings from detection task to key point detection task? Also I would prefer to add typing also in the docstrings.

Reply:
fixed at 713190f753fa04d865ba9a78ab58328a99517710